jubilant/_juju.py

```diff
@@ -837,6 +831,135 @@ def scp(
 
         self.cli(*args)
 
+    def secrets(self) -> list[Secret[Hidden]]:
+        """Get all secrets in the model.
+
+        Returns:
+            A list of :class:`Secret[Hidden]` objects, one for each secret in the model.
+        """
+        stdout = self.cli('secrets', '--format', 'json')
+        output: dict[str, SecretResponse] = json.loads(stdout)
+        return [
+            Secret[Hidden](
+                uri=SecretURI('secret:' + uri_from_juju),
+                name=obj.get('name', None),
+                owner=obj.get('owner', '<model>'),
+                rotation=obj.get('rotation', 'never'),
+                revision=obj.get('revision', 1),
+                description=obj.get('description', ''),
+                created=datetime.datetime.fromisoformat(
+                    str(obj.get('created', '').replace('Z', '+00:00'))
+                ),
+                updated=datetime.datetime.fromisoformat(
+                    str(obj.get('updated', '').replace('Z', '+00:00'))
+                ),
+                content=None,
+                checksum='',
+                expires=None,
+                rotates=None,
+                label=None,
+                error='',
+                access=[],
+                revisions=[],
```

Comment:
This has the same issue as `access`, although less concerning since `[]` is clearly an invalid value. However, I think `None` would make more sense here, or perhaps just copying the same solution as for `access`.